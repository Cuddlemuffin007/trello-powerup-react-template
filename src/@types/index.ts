export * from "./power-up";
export * from "./trello";
export * from "./types";
