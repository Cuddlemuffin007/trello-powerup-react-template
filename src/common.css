#react-root .ui.container.center.aligned,
#react-root .ui.segment.center.aligned {
  max-width: 280px!important;
  margin: 0!important;
}
