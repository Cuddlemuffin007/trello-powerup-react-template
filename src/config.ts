export default {
    trello: {
        appKey: process.env.POWERUP_APP_KEY,
        appName: process.env.POWERUP_NAME,
        tokenExpiration: process.env.TRELLO_TOKEN_EXPIRATION,
    },
};
