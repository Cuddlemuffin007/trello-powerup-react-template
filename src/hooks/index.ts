export * from "./useProvidedTrello";
export * from "./useTrelloApi";
