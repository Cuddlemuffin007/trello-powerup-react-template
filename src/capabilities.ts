import { Trello, CapabilityProps } from "./@types";
import { getCardButtons } from "./card-button/capability";

const CAPABILITY_PROPS: CapabilityProps = {
    baseUrl: window.location.href.replace(/\/$/, ""),
    icon: {
        dark: "/static/icon-dark.png",
        light: "/static/icon-light.png",
    },
};

export const connector = (): void => {
    window.TrelloPowerUp.initialize({
        "card-buttons": (t: Trello.PowerUp.IFrame) => getCardButtons(t, CAPABILITY_PROPS),
    });
};

/* istanbul ignore if */
if (typeof module !== undefined && !module.parent) {
    connector();
}
