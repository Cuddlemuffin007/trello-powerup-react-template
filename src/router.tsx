import React, { Suspense } from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { Segment } from "semantic-ui-react";

import { PowerUpRoutes } from "./components/PowerUpRoutes";
import { TrelloProvider } from "./components/TrelloProvider";
import { useTrelloApi } from "./hooks";

export const PowerUpRouter = (): JSX.Element => {
    const t = useTrelloApi();

    return (
        <TrelloProvider t={t}>
            <Suspense fallback={<Segment loading></Segment>}>
                <Router basename={process.env.CONTEXT_PATH}>
                    <PowerUpRoutes />
                </Router>
            </Suspense>
        </TrelloProvider>
    );
};
