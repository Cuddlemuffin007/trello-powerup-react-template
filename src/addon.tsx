import React from "react";
import * as ReactDOM from "react-dom";
import { PowerUpRouter } from "./router";
// import "semantic-ui-css/semantic.min.css";

ReactDOM.render(<PowerUpRouter />, document.getElementById("react-root"));
