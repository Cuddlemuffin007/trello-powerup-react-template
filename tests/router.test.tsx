import React from "react";
import { mount } from "enzyme";
import { waitFor } from "@testing-library/react";

import { PowerUpRouter } from "../src/router";
import { TrelloPowerUp } from "./helpers";

describe("application router", () => {
    it("should create a trello powerup iframe and render without crashing", async () => {
        const app = mount(<PowerUpRouter />);
        await waitFor(() => {
            expect(TrelloPowerUp.iframe).toHaveBeenCalledWith({
                appKey: "test_app_key",
                appName: "Trello PowerUp React",
            });
        });
        expect(app.update()).toMatchSnapshot();
        app.unmount();
    });
});
