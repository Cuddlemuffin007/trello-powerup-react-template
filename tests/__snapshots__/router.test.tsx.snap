// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`application router should create a trello powerup iframe and render without crashing 1`] = `
<PowerUpRouter>
  <TrelloProvider>
    <Suspense
      fallback={
        <Segment
          loading={true}
        />
      }
    >
      <BrowserRouter
        basename="/"
      >
        <Router
          basename="/"
          location={
            Object {
              "hash": "",
              "key": "default",
              "pathname": "/",
              "search": "",
              "state": null,
            }
          }
          navigationType="POP"
          navigator={
            Object {
              "action": "POP",
              "back": [Function],
              "block": [Function],
              "createHref": [Function],
              "forward": [Function],
              "go": [Function],
              "listen": [Function],
              "location": Object {
                "hash": "",
                "key": "default",
                "pathname": "/",
                "search": "",
                "state": null,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
        >
          <PowerUpRoutes>
            <Routes>
              <Message
                compact={true}
                content="Nothing to see here!"
                info={true}
              >
                <div
                  className="ui compact info message"
                >
                  <MessageContent>
                    <div
                      className="content"
                    >
                      <p>
                        Nothing to see here!
                      </p>
                    </div>
                  </MessageContent>
                </div>
              </Message>
            </Routes>
          </PowerUpRoutes>
        </Router>
      </BrowserRouter>
    </Suspense>
  </TrelloProvider>
</PowerUpRouter>
`;
